lib/china_regions/helpers/form_helper.rb

```diff
@@ -4,9 +4,9 @@ module Helpers
     module FormHelper
       def region_select(object_name, methods, options = {}, html_options = {})
         output = ''
-        preselected_choices = {}
+        @preselected_choices = {}
```

Comment:
But was that really a problem before? If anything, splitting it into two methods could reduce the number of arguments. I didn't think that was really a problem before, though, 4 arguments is the upper bound of "ok."

More important is knowing exactly what variables get changed where. With those instance variables, I have to follow the entire flow of the entire module to know when that variable gets set, which dramatically increases the complexity. I don't know when something is going to come in and change `@preselected_choices`. That's the beauty of local variables and encapsulation - I know what goes in, and what comes out. :)